Automattic/jetpack
# Launchpad: Enable to temporary dismiss a dismissible launchpad

Fixes part of #https://github.com/Automattic/wp-calypso/issues/85037

## Proposed changes:
* Update is_dismissed flag to consider temporary dismisses 
* Enable to set a dismissible date


## Does this pull request change what data or activity we track or use?
it only adds new values to already existent wp options.

### Other information:



## Testing instructions:
* Make sure you are sandboxed
* bin/jetpack-downloader test jetpack-mu-wpcom-plugin add/temporary-dismiss-to-launchpad
* Go to https://developer.wordpress.com/docs/api/console/ 
* Set GET (/sites/YOUR_SITE/launchpad) with the following params:
`checklist_slug:`   **intent-build**
`launchpad_context:`  **customer-home**
* Check if the results don't have a value on the dismissed_until value field
* Check if the flag 'is_dismissed' is false. 

* Open another tab with the https://developer.wordpress.com/docs/api/console/ 
* Set PUT (/sites/YOUR_SITE/launchpad) URL
* Set the is_checklist_dismissed field with the values:
``` json
{
  "slug":"intent-build",
  "dismiss_by": "+ 1 day"
}
```  
*  Use the get endpoint to check if the flag 'is_dismissed' was set to true
* Wait the 1 hour or hack the backend with an temporary date. 
* Use the get endpoint to check if the flag 'is_dismissed' was set to false.
